src/main/java/org/folio/circulation/resources/RequestByInstanceIdResource.java

```diff
@@ -109,31 +113,25 @@ private void createInstanceLevelRequests(RoutingContext routingContext) {
     final InstanceRequestRelatedRecords requestRelatedRecords = new InstanceRequestRelatedRecords();
     requestRelatedRecords.setInstanceLevelRequest(requestByInstanceIdRequest);
 
-    CompletableFuture<Result<InstanceRequestRelatedRecords>> itemsListsFuture =
-        finder.getItemsByInstanceId(requestByInstanceIdRequest.getInstanceId())
-              .thenApply(r -> r.next(items -> segregateItemsList(items, requestRelatedRecords)));
-
-    CompletableFuture<Result<InstanceRequestRelatedRecords>> sortedAvailableItemsFuture =
-        itemsListsFuture.thenApply(r -> r.next(RequestByInstanceIdResource::rankItemsByMatchingServicePoint));
-    CompletableFuture<Result<Map<Item, DateTime>>> loanItemsFuture =
-        itemsListsFuture.thenCompose(r -> r.after(relatedRecords -> getLoanItems(relatedRecords, clients)));
-    CompletableFuture<Result<Map<Item, Integer>>> requestQueuesFuture =
-        itemsListsFuture.thenCompose(r -> r.after(relatedRecords -> getRequestQueues(relatedRecords, clients)));
-
-    CompletableFuture.allOf(sortedAvailableItemsFuture, loanItemsFuture, requestQueuesFuture)
-                     .thenApply( x -> combineWithUnavailableItems(loanItemsFuture.join(), requestQueuesFuture.join(), requestRelatedRecords))
-                     .thenApply( r -> r.next(RequestByInstanceIdResource::instanceToItemRequests))
-                     .thenCompose( r -> r.after( requests -> placeRequests(requests, clients)))
-                     .thenApply(r -> r.map(RequestAndRelatedRecords::getRequest))
-                     .thenApply(r -> r.map(new RequestRepresentation()::extendedRepresentation))
-                     .thenApply(CreatedJsonResponseResult::from)
-                     .thenAccept(result -> result.writeTo(routingContext.response()))
-                     .exceptionally( err -> {
-                         String reason = "Error processing title-level request";
-                         log.error(reason, err);
-                         ServerErrorResponse.internalError(routingContext.response(), reason);
-                         return null;
-                      });
+    finder.getItemsByInstanceId(requestByInstanceIdRequest.getInstanceId())
+          .thenCompose(r -> r.after(items -> getRequestQueues(items,requestRelatedRecords, clients)))
+          .thenApply(r -> r.next(items -> validateRequester(items, requestByInstanceIdRequest.getRequesterId())))
```

Comment:
I guess the downside here is we are now doing everything in order. That's a shame. 😞

Reply:
Unfortunately